dask_geopandas/core.py

```diff
@@ -429,6 +432,100 @@ def geohash(self, string=True, p=12):
 
         return geohashes
 
+    def spatial_shuffle(
+        self,
+        by="hilbert",
+        p=None,
+        calculate_partitions=True,
+        npartitions=None,
+        divisions=None,
+        **kwargs,
+    ):
+        """
+        Shuffle the data into spatially consistent partitions.
+
+        This realigns the dataset to be spatially sorted, i.e. geometries that are
+        spatially near each other together will be within the same partition. This is
+        useful especially for overlay operations like a spatial join as it reduces the
+        number of interactions between individual partitions.
+
+        The spatial information is stored in the index and will replace the existing
+        index.
+
+        Note that ``spatial_shuffle`` uses ``set_index`` under the hood and comes with
+        all its potential performance drawbacks.
+
+        Parameters
+        ----------
+        by : string (default 'hilbert')
+            sorting method, one of {'hilbert', 'morton, 'geohash'}. See
+            ``hilbert_distance``, ``morton_distance`` and ``geohash`` methods for
+            details.
+        p : int (default None)
```

Comment:
Not for this PR, but I am wondering if we shouldn't rename this for all those methods to `precision` for clarity

Reply:
That is a good point.